spring-integration-jdbc/src/main/java/org/springframework/integration/jdbc/channel/PostgresChannelMessageTableSubscription.java

```diff
@@ -0,0 +1,48 @@
+package org.springframework.integration.jdbc.channel;
+
+import org.springframework.integration.jdbc.store.JdbcChannelMessageStore;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.MessageHandler;
+
+public final class PostgresChannelMessageTableSubscription {
```

Comment:
I don't think that I can accept this abstraction:
1. It is probably better to have a `JdbcChannelMessageStore` as a ctor for the subscriber.
2. Have getter for it over there as well
3. The Postgres message channel should expect the subscriber as its ctor and take that store from there. 
This way we would avoid the problem when subscriber and store look into different DBs or tables.
4. The subscription should be an inner interface of the subscriber to avoid classes tangle.
5. This way a `region` would be just an internal of the subscriber since both subscriber and channel would rely on the same store, therefore on the same region.
6. I don't believe in external `Runnable onPossibleUpdate`: it is really has to be an `onUpdate()` impl details. That's why I'm asking about a `SubscribableChannel` impl together with the mentioned subscription contract.
7. Somehow "subscriber" and "subscription" words don't work for me here: they are clashing with Reactive Streams. Not sure yet in other names though...

Reply:
Not sure if I follow, you suggest to make the JdbcChannelMessageStore a constructor argument to the subscriber? Then it would no longer be possible to reuse the subscriber for different regions. This would require multiple connections to block for each region, would it not? Ideally, I wanted to avoid this.

Reply:
Yeah... Sorry, I forgot that `JdbcChannelMessageStore` is restricted to specific region.
We will have to document it properly (and in JavaDocs as well), that `JdbcChannelMessageStore` on the channel we want to subscribe from and `PgConnectionSupplier` must belong to the same table.

Reply:
Yeah, I think this will need some extensive javadoc, but I wanted to get the API right before I started on writing anything there. Due to the sharing requirement, the API will get slightly more complicated than it otherwise would have been, but I think it's worth that performance-wise.